packages/lib/src/components/ApplePay/ApplePay.tsx

```diff
@@ -91,6 +95,28 @@ class ApplePayElement extends UIElement<ApplePayElementProps> {
         session.begin();
     }
 
+    private validateMerchant(resolve, reject) {
+        const { hostname: domainName } = window.location;
+        const { clientKey, configuration, loadingContext, initiative } = this.props;
+        const { merchantName: displayName, merchantIdentifier } = configuration;
+        const path = `${APPLEPAY_SESSION_ENDPOINT}?token=${clientKey}`;
+        const options = { loadingContext, path, method: 'post' };
+        const request: ApplePaySessionRequest = { displayName, domainName, initiative, merchantIdentifier };
+
+        return fetchJsonData(options, request)
```

Comment:
Maybe, since we already use a try-catch block we could use async/await here.

Reply:
Good idea 👍